pkg/util/stringreplace/object.go

```diff
@@ -0,0 +1,53 @@
+package stringreplace
+
+import (
+	"reflect"
+
+	"github.com/golang/glog"
+)
+
+// VisitObjectStrings visits recursively all string fields in the object and call the
+// visitor function on them. The visitor function can be used to modify the
+// value of the string fields.
+func VisitObjectStrings(obj interface{}, visitor func(string) string) {
+	v := reflect.ValueOf(obj).Elem()
+
+	visitField := func(val reflect.Value) {
+		switch val.Kind() {
+		case reflect.Ptr, reflect.Interface:
+			if val.CanInterface() {
+				VisitObjectStrings(val.Interface(), visitor)
+			}
+		case reflect.Struct, reflect.Map, reflect.Slice, reflect.Array:
+			if val.CanInterface() {
+				VisitObjectStrings(val.Addr().Interface(), visitor)
+			}
+		case reflect.String:
+			if !val.CanSet() {
+				glog.V(5).Infof("Unable to set String value '%v'", v)
+			}
+			val.SetString(visitor(val.String()))
+		}
+	}
+
+	switch v.Kind() {
+	case reflect.Slice, reflect.Array:
+		for c := 0; c < v.Len(); c++ {
+			visitField(v.Index(c))
+		}
+	case reflect.Struct:
+		for i := 0; i < v.NumField(); i++ {
+			visitField(v.Field(i))
+		}
+	case reflect.Map:
+		for _, k := range v.MapKeys() {
```

Comment:
not certain there's a strong use case, but what about substituting keys in maps, not just values?